evm-tests/jsontests/src/state.rs

```diff
@@ -559,13 +559,31 @@ fn test_run(
 							let code = data;
 							let value = transaction.value.into();
 
-							let _reason = executor.transact_create(
+							let reason = executor.transact_create(
 								caller,
 								value,
 								code,
 								gas_limit,
 								access_list,
 							);
+							if let Some(expect_exception) = state.expect_exception.as_ref() {
```

Comment:
@birchmd it looks like it doesn't work like expected.
For tests:
```
 [Cancun]  CREATE_ContractRETURNBigOffset:0
 [Cancun]  CREATE_ContractRETURNBigOffset:1
 [Cancun]  CREATE_ContractRETURNBigOffset:2
 [Cancun]  CREATE_ContractRETURNBigOffset:3
 [Cancun]  randomStatetest643:0
 [Cancun]  codesizeOOGInvalidSize:0
 [Cancun]  codesizeOOGInvalidSize:1

```

it does not contain `state.expect_exception`, and `ExitReason` does not check for results and tests passed successfully.

Field `state.expect_exception` for 20000 tests is only for 2 tests.

Reply:
Added refactoring in [fda0b97](https://github.com/aurora-is-near/sputnikvm/pull/41/commits/fda0b977cf2e7b7ca70e4d73aebce9c794a1ee58)

Reply:
You're saying more tests fail when you always look for `ExitError::CreateContractLimit` instead of only when `state.expect_exception` is non-empty? Doesn't that mean there is something wrong with those other tests?

Reply:
But what criteria for tests passing? `json-eth-tests` do not contain any information about EVM exit status. Only one criterion is state hash assertion.

Reply:
After additional investigation (including REVM tests solutions), I found that sometimes `ExitReason::Error` is really the expected result for tests. And it's a lot of tests with non-success results.

And for that cases (in REVM for example) checked is output empty.

Reply:
So as solution I can extend `ExitReason` error flow check with non-empty output.

Reply:
Got it, thanks for the clarification. I suppose the current code is fine as-is since `state.expect_exception` is not consistently used.